pkg/booking/booking.go

```diff
@@ -86,7 +91,16 @@ func (bs *Service) MintBookingToken(
 	if strings.TrimSpace(uri) == "" {
 		return nil, fmt.Errorf("uri cannot be empty")
 	}
-
+	if paymentToken != zeroAddress {
+		erc20Contract, err := erc20.NewErc20(paymentToken, bs.client)
```

Comment:
I'd suggest to add lru cache for bindings, there probably will be just a few ones that will be used often. Create new struct that will contain that cache, init it in app and pass this struct to both booking and responseHandler